src/rlx_resolve.erl

```diff
@@ -44,34 +44,49 @@ solve_release(Release, State0) ->
                         {false, [SystemLibs | LibDirs]}
                 end,
 
-            Pkgs = subset(Goals, AllApps, LibDirs1, CheckCodeLibDirs),
-            Pkgs1 = remove_exclude_apps(Pkgs, State1),
-            set_resolved(Release, Pkgs1, State1)
+            ExcludeApps = rlx_state:exclude_apps(State1),
+            Pkgs = subset(Goals, AllApps, LibDirs1, CheckCodeLibDirs, ExcludeApps),
+            set_resolved(Release, Pkgs, State1)
     end.
 
 %% find the app_info records for each application and its deps needed for the release
-subset(Apps, World, LibDirs, CheckCodeLibDirs) ->
-    subset(Apps, World, sets:new(), LibDirs, CheckCodeLibDirs, []).
+subset(Goals, World, LibDirs, CheckCodeLibDirs, ExcludeApps) ->
+    {Apps, _} = fold_apps(Goals, World, sets:new(), LibDirs, CheckCodeLibDirs, ExcludeApps),
+    Apps.
 
-subset([], _World, _Seen, _LibDirs, _CheckCodeLibDirs, Acc) ->
-    Acc;
-subset([Goal | Rest], World, Seen, LibDirs, CheckCodeLibDirs, Acc) ->
+subset(Goal, World, Seen, LibDirs, CheckCodeLibDirs, ExcludeApps) ->
     {Name, Vsn} = name_version(Goal),
     case sets:is_element(Name, Seen) of
         true ->
-            subset(Rest, World, Seen, LibDirs, CheckCodeLibDirs, Acc);
+            {[], Seen};
         _ ->
             AppInfo=#{applications := Applications,
                       included_applications := IncludedApplications} =
                 find_app(Name, Vsn, World, LibDirs, CheckCodeLibDirs),
-            subset(Rest ++ Applications ++ IncludedApplications,
-                   World,
-                   sets:add_element(Name, Seen),
-                   LibDirs,
-                   CheckCodeLibDirs,
-                   Acc ++ [AppInfo])
+            {Apps, Seen2} = fold_apps(Applications ++ IncludedApplications,
+                                      World,
+                                      sets:add_element(Name, Seen),
+                                      LibDirs,
+                                      CheckCodeLibDirs,
+                                      ExcludeApps),
+
+            %% don't add excluded apps
+            %% TODO: should an excluded apps also be excluded?
```

Comment:
what does this comment mean? Was it "Should an excluded app's deps also be excluded" ? I guess the answer would be "yes, but only if nothing else depends on it" ? Probably no big deal for now since those can also just manually be excluded fine.

Reply:
Yup, fixed the comment.

And if I were adding it today I would have it also exclude the deps, but the feature already exists, and has for a while, so I don't know that we can change it. 

reltool has a similar feature, I guess I need to check how it does it.